bijection-hbase/src/test/scala/com/twitter/bijection/hbase/HBaseInjectionsSpecifications.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.twitter.bijection.hbase
+
+import scala.util.Success
+import com.twitter.bijection.Injection
+import com.twitter.bijection.hbase.HBaseInjections._
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable
+import org.scalatest.WordSpec
+
+/**
+ * @author Muhammad Ashraf
+ * @since 7/10/13
+ */
+class HBaseInjectionsSpecifications extends WordSpec {
+  "HBaseInjections" should {
+    "respect ImmutableBytesWritable offset and length" in {
+      val long: Long = 1
+      val double: Double = 2
+      val int: Int = 3
+
+      val longIBW = long2BytesWritableInj(long)
+      val doubleIBW = double2BytesWritableInj(double)
+      val intIBW = int2BytesWritableInj(int)
+
+      val allBytes = longIBW.copyBytes ++ doubleIBW.copyBytes ++ intIBW.copyBytes
+
+      val sharedLongIBW = new ImmutableBytesWritable(allBytes, 0, 8)
+      val sharedDoubleIBW = new ImmutableBytesWritable(allBytes, 8, 8)
+      val sharedIntIBW = new ImmutableBytesWritable(allBytes, 16, 4)
+
+      assert(Success(long) == long2BytesWritableInj.invert(sharedLongIBW))
+      assert(Success(double) == double2BytesWritableInj.invert(sharedDoubleIBW))
+      assert(Success(int) == int2BytesWritableInj.invert(sharedIntIBW))
+    }
+
+    "round trip String -> Array[Byte]" in {
```

Comment:
This test ensures that, in a situation where 3 different values are backed by different parts of the same byte array, the hbase injection inversions still work.